test/func_memory_hma.py

```diff
@@ -106,22 +106,348 @@ def memory_hma_alloc(self):
 
 """)
 
-    results = self.runDosemu("testit.bat", config=config)
+    results = self.runDosemu("testit.bat", config=CONFIG)
+
+    self.assertIn("PASS:", results)
+
+
+def memory_hma_alloc3(self):
+
+    if 'FDPP' in self.version:
+        self.mkfile("userhook.sys", """DOS=HIGH\n""", newline="\r\n")
+
+    self.mkfile("testit.bat", BATCHFILE % 'hmaalloc', newline="\r\n")
+
+    self.mkcom_with_ia16("hmaalloc", r"""
+
+#include <i86.h>
+#include <stdio.h>
+#include <string.h>
+
+struct HMCB {
+  uint8_t signature[2]; // "MS"
+  uint16_t owner;       // 0000=free, 0001=DOS, FF33=IO.SYS, FFFF=MSDOS.SYS
+  uint16_t size;        // bytes not including this header
+  uint16_t next;        // offset of next memory block in segment FFFFh, or 0000h if last
+  uint8_t reserved[8];  // unused (explicitly set to 0 for MS-DOS 7.10)
+};
+
+/*
+Windows95 - DOS KERNEL - (DE)ALLOCATE HMA MEMORY BLOCK
+
+AX = 4A03h
+CX = segment of block's owner???
+DL = subfunction
+  00h allocate block
+    BX = number of bytes
+    Return:
+      DI=FFFFh if unable to allocate
+      ES:DI -> allocated block
+      // We do get the allocated size in BX, RBIL misses this
+
+  01h resize block // Seemingly can grow as well as shrink
+    ES:DI -> previously-allocated block
+    BX = new size in bytes
+    Return:
+      DI=FFFFh if unable to allocate
+      ES:DI -> reallocated block
+      // Maybe we get the reallocated size in BX
+  Win95 FE & SE never seem to free the initial block after creating the
+  replacement. In addition, subsequent frees of the replacement block
+  fail.
+
+  02h free block
+    ES:DI -> block to be freed
+    Return:
+      Nothing, but checking the ID in the requested block can indicate
+      whether the block was freed.
+*/
+
+int main(int argc, char *argv[])
+{
+  union REGS r;
+  struct SREGS rs;
```

Comment:
You should use `= {}` in such
cases to not pass garbage in
other registers. Its especially
important to initialize flags to
not get TF set.
But I'll leave that for another
patch of course.